apps/extension/src/contents/inpage.ts

```diff
@@ -49,7 +49,7 @@ export const windowAI: WindowAI<ModelID> = {
         requestId,
         (res) => {
           if (isOk(res)) {
-            resolve(res.data)
+            resolve(shouldReturnMultiple ? res.data : (res.data[0] as any))
```

Comment:
Wow, interesting that this doesn’t force the result type to be “any” (right?). Is it true that there’s a tricky edge case here where if numOutputs is set to 1, then the return type of getCompletion will be an array, but the return value will be a single Output?

If so maybe it’s best to bite the bullet here and just always return an array from the lib only…

Reply:
> Wow, interesting that this doesn’t force the result type to be “any” (right?)

The return type of this Promise is inferred by the interface but the type assertion on `options` doesn't affect the return type of the Promise at the implementation level unfortunately... - i.e it can do a type-guard to assert options contains numsOutput, but only the type of `options` is inferred, not the type of the interface's promise result. 

>  Is it true that there’s a tricky edge case here where if numOutputs is set to 1, then the return type of getCompletion will be an array, but the return value will be a single Output?

Yup lol

> If so maybe it’s best to bite the bullet here and just always return an array from the lib only…

Another way to reduce confusion is to remove the `numOuputs > 1` check on the extension side. Since by default it's 1, people wouldn't specify it anyway. But for folks who might want to experiment with an array of output, having a consistent array type would help them bump the number accordingly.